### ISSUE TITLE
PreparedStatements seems never get closed

### ISSUE THREAD

**Issue by jilen** [mauricio/postgresql-async#195]
Which will leads to #194 .   As a normal jdbc driver, we must call `statment.close` after using an prepared statement

**Comment by mauricio** [comment]
Yeah, I started work for this at https://github.com/mauricio/postgresql-async/tree/closing-prepared-statements but the MySQL piece is giving me trouble, any help is appreciated.

**Comment by jilen** [comment]
@mauricio I am trying to close it like [this](https://github.com/jilen/postgresql-async/blob/0.2.20-DRIP/mysql-async/src/main/scala/com/github/mauricio/async/db/mysql/codec/MySQLConnectionHandler.scala#L63). But actually with no confidence whether it is correctly

**Comment by Narigo** [comment]
@mauricio I've looked into the code a bit, what do you think about a different API for releasing prepared statements?
If I understand the code in `Connection` correctly you want to do this:

```scala
val sql = "SELECT something FROM t WHERE id=?"
connection.sendPreparedStatement(sql, List(1))
  .flatMap(doSomething)
  .flatMap(_ => connection.releasePreparedStatement(sql))
```

I'd propose to let `Connection.sendPreparedStatement` return a `Future[PreparedStatementQueryResult]`, where `PreparedStatementQueryResult` is something like this:
```scala
class PreparedStatementQueryResult(val rowsAffected: Long, val statusMessage: String, val rows: Option[ResultSet] = None) extends QueryResult(rowsAffected, statusMessage, rows) {
  def release(): Future[Boolean] = ???
}
```

What do you think about that?

**Comment by Narigo** [comment]
I just saw that this issue is more or less a duplicate of #111 @mauricio :)

**Comment by mauricio** [comment]
Yup, MySQL is still a pain in this :(

**Comment by oshai** [comment]
I merged the fix in jasync fork: https://github.com/jasync-sql/jasync-sql/issues/82
